### PR TITLE
Diagram widget data binding

### PR DESCRIPTION
**Pull request by Romzay**

### CODE REVIEW

File: js/ui/diagram/ui.diagram.js [hunk 1]
```diff
@@ -89,26 +94,118 @@ class Diagram extends Widget {
...
+        }
+    }
+    _isBindingMode() {
+        return this._nodes || this._edges;
+    }
+    _beginUpdateDiagram() {
+        this._updateDiagramLockCount++;
+    }
+    _endUpdateDiagram() {
+        this._updateDiagramLockCount = Math.max(this._updateLockCount - 1, 0);
```
Comment (maxerokh) [inline js/ui/diagram/ui.diagram.js]: I couldn't found mention about the this._updateLockCount variable.
What about the `Math.max(this. _updateDiagramLockCount - 1, 0);` code?

File: js/ui/diagram/ui.diagram.js [hunk 2]
```diff
@@ -89,26 +94,118 @@ class Diagram extends Widget {
...
+                toField: this.option("edges.toExpr")
+            },
+            layoutType: this._getDataLayoutType()
+        };
+        this._diagramInstance.commandManager.getCommand(DiagramCommand.BindDocument).execute(data);
+    }
+    _getDataLayoutType() {
+        const { DataLayoutType } = getDiagram();
+        switch(this.option("layout")) {
+            case "tree":
```
Comment (maxerokh) [inline js/ui/diagram/ui.diagram.js]: You may remove the `case "tree":` code. Default case will handle this situation too.

**Review by maxerokh — CHANGES_REQUESTED** [review]

**Review by maxerokh — DISMISSED** [review]

**Review by maxerokh — APPROVED** [review]

File: js/docEnums.js [hunk 3]
```diff
@@ -1015,3 +1015,8 @@
  * @typedef {string} Enums.FileManagerItemViewMode
  * @enum {'details'|'thumbnails'}
  */
+
+/**
+ * @typedef {string} Enums.DiagramAutoLayout
```
Comment (San4es) [inline js/docEnums.js:1020]: /cc @RomanTsukanov

**Review by San4es — APPROVED** [review]
LGTM

File: js/docEnums.js (hunk 3 above)
Comment (RomanTsukanov) [reply js/docEnums.js:1020]: @Romzay Why do we need `Auto` in the name?

**Pull request merged**